packages/odata-common/src/request-builder/batch/batch-response-deserializer.ts

```diff
@@ -68,21 +68,25 @@ export class BatchResponseDeserializer<DeSerializersT extends DeSerializers> {
         this.responseDataAccessor,
         this.deserializer
       ),
-      isSuccess: () => true,
-      isError: () => false,
-      isReadResponse: () => true,
-      isWriteResponses: () => false
+      isSuccess: (): this is
```

Comment:
I assume TS 5.5.4 enforces stricter checks on predicates compared to 5.4. The error here was: 

```
Type '() => true' is not assignable to type '() => this is ReadResponse<DeSerializersT> | WriteResponses<DeSerializersT>'.
  Signature '(): true' must be a type predicate.ts(2322)
batch-response.ts(15, 3): The expected type comes from property 'isSuccess' which is declared here on type 'ReadResponse<DeSerializersT>'
```